### PR TITLE
Update LinkedIn to use new lite profile

### LINKED ISSUE #390: Upgrade LinkedIn to support new 'Lite Profile'
Couple small changes LinkedIn is making that need to be accounted for. Mainly:

 - r_liteprofile (replaces r_basicprofile)
 - email no longer part of the profile request

https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/migration-faq?context=linkedin/consumer/context

### PR DESCRIPTION
**Pull request by dawnerd**
### Changes
 - Switched to use the new `liteprofile` scope
 - Grabs email from new dedicated endpoint

Will be testing this in another app tomorrow just to make sure I didn't miss anything.

Note: I had to update mock to support overriding both the endpoints - not sure if thats the best way...

Resolves #390

### CODE REVIEW

**Comment by dawnerd** [comment]
@AdriVanHoudt Can we get some eyes on this soon?

**Comment by dawnerd** [comment]
Got around to trying it out and had to make a fix, looks like they're using standard oauth now. Their docs are also a bit wrong with getting the email.

File: lib/providers/linkedin.js [hunk 1]
```diff
@@ -15,34 +13,31 @@ exports = module.exports = function (options) {
...
                 name: {
-                    first: profile.firstName,
-                    last: profile.lastName
+                    first: profile.firstName.localized[Object.keys(profile.firstName.localized)[0]],
+                    last: profile.lastName.localized[Object.keys(profile.lastName.localized)[0]]
                 },
-                email: profile.emailAddress,
-                headline: profile.headline,
-                raw: profile
+                email: email.elements[0]['handle~'].emailAddress,
```
[the remark this answers is not in the document]
Comment (dawnerd) [reply lib/providers/linkedin.js]: Yeah that's what I referred to about their docs being wrong. They send back an elements array.

File: test/mock.js [hunk 2]
```diff
@@ -258,7 +258,12 @@ exports.override = function (uri, payload) {
 
             if (dest.indexOf(uri) === 0) {
                 if (typeof payload === 'function') {
-                    await payload(dest);
+                    const res = await payload(dest);
+
+                    if (res) {
+                        return { res: { statusCode: 200 }, payload: JSON.stringify(res) };
```
[the remark this answers is not in the document]
Comment (dawnerd) [reply test/mock.js]: Fixed it locally, but not sure where `Expected 'https://api.linkedin.com/v2/me' to equal specified value: 'https://api.linkedin.com/v1/people/~'` is coming from. Ideas?

[the remark this answers is not in the document]
Comment (AdriVanHoudt) [reply test/mock.js]: Strange, on master there https://github.com/hapijs/bell/blob/master/test/providers/linkedin.js#L117
Which I added here https://github.com/hapijs/bell/commit/b7e909bd62f3c1cdff475706d4dd28f6b369492b#diff-3c85fecb7fa29dc987a4a01cf9b48aa8 to get coverage up due to fixes in Lab
Maybe rebase/merge master into this branch and check again?

File: lib/providers/linkedin.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AdriVanHoudt) [reply lib/providers/linkedin.js]: That's... something haha

File: test/mock.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dawnerd) [reply test/mock.js]: That was it!

**Pull request merged**